lib/config/config-file.js

```diff
@@ -176,6 +176,36 @@ function loadPackage(filePath) {
 }
 
 /**
+ * Loads a JavaScript configuration from a plugin.
+ * @param {string} filePath The plugin name to load.
+ * @returns {Object} The configration object from the plugin.
+ * @throws {Error} If the package cannot be read.
+ * @private
+ */
+function loadPlugin(filePath) {
+    var parts = filePath.split("/"),
+        pluginName = parts[0],
+        configName = parts[1];
```

Comment:
If I'm reading this right, there's a way to supply a default config :smile:

If `filePath` is `"eslint-plugin-myplugin"` (without specifying any config name), then `pluginName` will be `"eslint-plugin-myplugin"`, and `configName` will be `undefined`. Below, it will check for `plugin.config[undefined]`, which is the same as `plugin.config["undefined"]`, so a package could supply its "default" config under the name `"undefined"`.

Sadly I suspect this is unintentional and might be fixed, which is much less fun :disappointed: